core/src/main/scala/kafka/controller/TopicDeletionManager.scala

```diff
@@ -88,6 +86,10 @@ class TopicDeletionManager(controller: KafkaController,
   val deleteTopicStateChanged: AtomicBoolean = new AtomicBoolean(false)
   var deleteTopicsThread: DeleteTopicsThread = null
   val isDeleteTopicEnabled = controller.config.deleteTopicEnable
+  val topicsToBeDeleted: mutable.Set[String] = mutable.Set.empty[String]
+  if (isDeleteTopicEnabled)
+    topicsToBeDeleted ++= initialTopicsToBeDeleted
+  val partitionsToBeDeleted: mutable.Set[TopicAndPartition] = topicsToBeDeleted.flatMap(controllerContext.partitionsForTopic)
```

Comment:
It seems that on controller failover, if delete topic is not enabled and there are topicsToBeDeleted, in addition to ignore those topics, we should remove them from the delete topic path.